### PR TITLE
[DDW-452] Show ada amounts in other currencies

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implements the display of ada amounts in other currencies.

## Todos

- [x] Research API's
- [x] Create an API agnostic interface, so we can plug & unplug different API's
- [x] Implement an initial status check, so the feature can be disabled if failed
- [x] Implement UI
- [x] Integrate UI and API
- [x] Copy & translation

## Screenshots

Screenshot 2021-02-12 at 09 56 41
Screenshot 2021-02-12 at 09 56 36
Screenshot 2021-02-12 at 09 59 24
Screenshot 2021-02-12 at 09 59 28
Screenshot 2021-02-12 at 09 58 50
Screenshot 2021-02-12 at 09 59 03
Screenshot 2021-02-12 at 09 59 18
Screenshot 2021-02-12 at 09 59 10

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

## Test scenarios

**Scenario 1 - Validate "Wallets" option is visible in "Settings"**
```gherkin
Given that I am on the "Settings" screen
When I check the available tabs
Then I can see that the "Wallets" option is visible  
```
**Scenario 2 - Validate "Wallets" tab**
```gherkin
Given that I am on the "Wallets" tab on the "Settings" screen
When I validate its content
Then I can see a brief description of the feature 
And a switch to enable or disable it
And by default, it's enabled
And "US Dollar" is the default option in the dropdown
```
**Scenario 3 - Disable displaying ada on other currencies**
```gherkin
Given that a wallet is synced on Daedalus
And I am on the "Wallets" tab on the "Settings" screen
When I click on the switch to disable the feature
Then I can see the dropdown and disclaimer are hidden
And on the "Summary" tab for a wallet, no other currency is displayed
```
**Scenario 4 - Enable displaying ada on other currencies and check** balance of a syncing wallet
```gherkin
Given that a wallet is syncing on Daedalus
And the option of displaying ada in other currencies is enabled
When I validate the balance of the wallet in the "Summary" tab
Then I can see the estimated conversion is displayed as "-"
```
**Scenario 5 - Enable displaying ada on other currencies and check** balance of a synced wallet
```gherkin
Given that a wallet is synced on Daedalus
And the option of displaying ada in other currencies is enabled
When I validate the balance of the wallet in the "Summary" tab
Then I can see the balance is displayed in ada
And to the right, the balance is displayed on the selected currency
```
**Scenario 6 - Access to "Wallets" tab from "Summary" tab**
```gherkin
Given that a wallet is synced on Daedalus
And the option of displaying ada in other currencies is enabled
And I am on the "Summary" tab of the wallet
When I click on the gear icon below the balance to the right
Then I'm redirected to the "Wallets" tab on the "Settings" screen
```
**Scenario 7 - Validate selection in "Wallets" tab is persisted after restart**
```gherkin
Given that I am on the "Wallets" tab on the "Settings" screen
When I enable or disable the feature
And I restart Daedalus
Then I can see that the change was persisted
```
**Scenario 8 - Validate CoinGecko link**
```gherkin
Given that I am on the "Wallets" tab on "Settings" screen
And the option of displaying ada in other currencies is enabled
When I click on the link "CoinGecko"
Then the foloowing link opens in the browser:

https://www.coingecko.com/en/api
```

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by matiwinnetou** [comment]
There is also onchain metadata with currency data provided by me and stakepool: NUTS. 

https://nut.link/

**Comment by nikolaglumac** [comment]
> There is also onchain metadata with currency data provided by me and stakepool: NUTS.
> 
> https://nut.link/

Thanks for the tip @matiwinnetou - much appreciated :-)

**Comment by darko-mijic** [comment]
So excited to give this one a spin. 😺

**Comment by darko-mijic** [comment]
@daniloprates, please add theming adjustents to todo list:
 
image

**Comment by darko-mijic** [comment]
@daniloprates, can you check if the refreshing of the UI component can be smoother? It removes the converted value before displaying the updated one. Maybe we should animate this somehow? @a-rukin?

**Comment by a-rukin** [comment]
image
image

@daniloprates can you make this "powered by" and "description" to be light font (now it's regular)

**Comment by a-rukin** [comment]
image
something with centering here

**Comment by a-rukin** [comment]
image
@daniloprates hovering should make this one nontransparent. Maybe there is some alpha inside svg already?

**Comment by a-rukin** [comment]
image
Here is extra "g" in name

**Comment by a-rukin** [comment]
> @daniloprates, can you check if the refreshing of the UI component can be smoother? It removes the converted value before displaying the updated one. Maybe we should animate this somehow? @a-rukin?

Can we not hide values while fetching?

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@daniloprates link to "CoinGecko" is not working - instead of opening the link in a browser, I get the dropdown focused...

Screenshot 2021-02-11 at 19 47 39

File: source/renderer/app/api/utils/localStorage.js [hunk 1]
```diff
@@ -181,6 +186,27 @@ export default class LocalStorageApi {
...
+    LocalStorageApi.set(keys.CURRENCY_SELECTED, currency);
+
+  unsetCurrencySelected = (): Promise<void> =>
+    LocalStorageApi.unset(keys.CURRENCY_SELECTED);
+
+  getCurrencyIsActive = (): Promise<boolean> =>
+    LocalStorageApi.get(keys.CURRENCY_ACTIVE, CURRENCY_IS_ACTIVE_BY_DEFAULT);
+
+  unsetCurrencyIsActive = (): Promise<void> =>
+    LocalStorageApi.unset(keys.CURRENCY_SELECTED);
```
Comment (nikolaglumac) [inline source/renderer/app/api/utils/localStorage.js:202]: @daniloprates shouldn't we here unset `keys.CURRENCY_ACTIVE`?

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@daniloprates I have left only one comment in the code. Please check it out and also don't forget to add screenshots 🙏 
Thanks! Awesome work 💯

**Comment by gnpf** [comment]
@daniloprates A few issues on build 16440:

1 - When the language is Japanese, the text next to the gear is displayed in English. Check [this](https://drive.google.com/file/d/1wfjCac4t6NmXV615EeVbRgrvQqeMd4rv/view?usp=sharing) screenshot.

2 - The balance is displayed as 0.00 while syncing. Check [this](https://drive.google.com/file/d/1spLAEfR2QR8G1s-LYR_Q5YetvgfDQsM3/view?usp=sharing) screenshot. Maybe this should also display "-" while the wallet is syncing?

3 - If I disable the option of seeing other currencies, it would be enabled again after a restart. Check [this](https://drive.google.com/file/d/1L_GHHK7x2PrdaGPIkxVqwBW2G-agfG7h/view?usp=sharing) video.

I'm adding some test scenarios to this PR, could you review them, please?

**Comment by nikolaglumac** [comment]
We need to implement some more text copy changes: https://docs.google.com/document/d/1YrqW0yW3O4p7bARMERDYK3Pz5a9bRIFMUQ5GZ0eaR5w/edit#heading=h.jo1dsl3gyz08

**Comment by nikolaglumac** [comment]
> We need to implement some more text copy changes: https://docs.google.com/document/d/1YrqW0yW3O4p7bARMERDYK3Pz5a9bRIFMUQ5GZ0eaR5w/edit#heading=h.jo1dsl3gyz08

Implemented in: https://github.com/input-output-hk/daedalus/pull/2290/commits/933a4d0d0b385b4349dddc1944fa24cb48c2aa5b

**Comment by nikolaglumac** [comment]
> @daniloprates A few issues on build 16440:
> 
> 1 - When the language is Japanese, the text next to the gear is displayed in English. Check [this](https://drive.google.com/file/d/1wfjCac4t6NmXV615EeVbRgrvQqeMd4rv/view?usp=sharing) screenshot.
> 
> 2 - The balance is displayed as 0.00 while syncing. Check [this](https://drive.google.com/file/d/1spLAEfR2QR8G1s-LYR_Q5YetvgfDQsM3/view?usp=sharing) screenshot. Maybe this should also display "-" while the wallet is syncing?
> 
> 3 - If I disable the option of seeing other currencies, it would be enabled again after a restart. Check [this](https://drive.google.com/file/d/1L_GHHK7x2PrdaGPIkxVqwBW2G-agfG7h/view?usp=sharing) video.
> 
> I'm adding some test scenarios to this PR, could you review them, please?

@gnpf I can confirm that all these issues have been fixed in the latest build.

**Review by nikolaglumac — APPROVED** [review]
Amazing job @daniloprates 🎉

**Comment by a-rukin** [comment]
@daniloprates this block requires "line-height:22px;" for explanation text 
without
image
with
image

**Comment by a-rukin** [comment]
image
"Converts as" too far to the top now

**Comment by a-rukin** [comment]
image
this padding should be 20px (can't diagnost why it adds a bit more)

**Review by ManusMcCole — APPROVED** [review]
Great job @daniloprates 👍

**Review by mioriohk — APPROVED** [review]
LGTM. Tested on 16454. Great job @daniloprates 👍

**Review by a-rukin — APPROVED** [review]
🧙

**Pull request merged**